src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,21 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String mark;
+    private String type;
+    private int maxSpeed;
+    private String color;
+    private Engine nameEngine;
+
+    @Override
+    public Car clone() {
+        Car clonedCar = new Car();
+        clonedCar.mark = this.mark;
+        clonedCar.type = this.type;
+        clonedCar.maxSpeed = this.maxSpeed;
+        clonedCar.color = this.color;
+        clonedCar.nameEngine = this.nameEngine;
```

Comment:
Do we need this keywords here?How about create a constructor that would get all fields and use it?

Reply:
Done